### ISSUE TITLE
gcc 8.3.0 high memory usage

### ISSUE THREAD

**Issue by kodidu** [redboltz/mqtt_cpp#901]
Hello!

Compiling the simple subscriber example in may application consumes additional 4GB+ RAM when compiling with gcc version 8.3.0.
I could also find #522 about that. But simply reducing the parallel build is no solution.
Last time I tested this lib v9.0.0 was used.

Was this issue addressed between v9.0.0 and v12.0.0?

**Comment by redboltz** [comment]
No. But some of updates could change memory usage.
I think that 4GB is too small. mqtt_cpp doesn't focus on reducing compile time memory usage, so far.
I recommend that add memory on compiler environment. If it is embedded environment, use cross compiler.

clang++ might use smaller memory, I'm not sure.

**Comment by Bart97** [comment]
I'm very sad to say that at this point it seems that the library is not usable in any complex projects.... It's a shame because I really liked this approach and the use of boost.asio, but the insane compilation times and memory usage make it a pain to work with.

**Comment by kodidu** [comment]
Yes it is sad, I also never experienced such bad compile-time memory / cpu usage with any library. Template/header-only libraries are hard to master.
